src/data/scripts/generate_historic_stats.py

```diff
@@ -0,0 +1,144 @@
+"""
+to run this script
+python3 -m src.data.scripts.generate_historic_stats
+from the root
+"""
+
+import json
+import pandas
+from src.data.scripts.utils import get_and_clean_csv, get_data_file_path, log_step_completion, output_to_csv
+
+debug = False
+
+# Write out the data
+output_filename = 'historic-stats.json'
+
+# Assume run in /data
+data_directory = 'source'
+data_out_directory = 'dist'
+
+# The /debug directory is to have a well formatted JSON for reading
+data_debug_directory = 'debug'
+
+# The input file from the previous step. Gatsby doesn't like spaces so we use a CSV with renamed
+# headers with no units
+building_emissions_file = 'benchmarking-all-years.csv'
+
+# The final output file name
+building_emissions_file_out_name = 'building-benchmarks'
+
+# Columns we want to run statistical analysis and ranking on - order matters here
+building_cols_to_analyze = [
+    'GHGIntensity',
+    'TotalGHGEmissions',
+    'ElectricityUse',
+    'NaturalGasUse',
+    'SourceEUI',
+    'SiteEUI',
+    'DistrictSteamUse',
+    'DistrictChilledWaterUse'
+]
+
+# Calculates year-over-year stats for all buildings and outputs them into a keyed JSON file
+def calculateBuildingStatsByYear(building_data_in: pandas.DataFrame) -> str:
+    """
+    Calculates year-over-year stats for all buildings and outputs them into a keyed JSON file.
+    Returns statistics for each year in the dataset.
+
+    Returns an array of files written to
+    """
+
+    # Clone the input data to prevent manipulating it on accident
+    building_data = building_data_in.copy()
+
+    # The details columns we want to keep. Note that 50% = median
+    detail_cols_to_keep = ['count', 'mean', 'std', 'min', 'max', '25%', '50%', '75%']
+
+    # Get all unique years in the dataset
+    years = sorted(building_data['DataYear'].unique())
+    
+    # Dictionary to store results for each year
+    yearly_stats = {}
+    
+    for year in years:
+                
+        if year < 2016: 
+          continue
+            
+        
+        # Filter data for this specific year
+        year_data = building_data[building_data['DataYear'] == year]
+        
+        # Calculate statistics for this year
+        year_stats_df = year_data[building_cols_to_analyze].describe().loc[detail_cols_to_keep]
+
+        
+        # Round all data to 1 decimal place
+        year_stats_df = year_stats_df.round(1)
+        
+        if debug:
+          print("-" * 60)
+          print(year)
+          print(year_stats_df)
+        
+        # Rename columns to work with GraphQL (no numbers like '25%')
+        year_stats_df.rename(index={
+            '25%': 'twentyFifthPercentile',
+            '50%': 'median',
+            '75%': 'seventyFifthPercentile',
+        }, inplace=True)
+        
+        # Convert to dictionary and store
+        yearly_stats[str(year)] = year_stats_df.to_dict()
+
+    # Create summary statistics across all years
```

Comment:
This comment says this does all years, but for me locally I just saw one 🤔 

<img width="1020" height="468" alt="image" src="https://github.com/user-attachments/assets/043b7962-b52d-4dc4-95ba-0d5dcf7a0519" />